File: .claude/hooks/check-nav-consistency.mjs
// ===========================================================
// Garde-fou : cohérence nav / footer entre les 3 pages HTML
// (règle CLAUDE.md : "tout changement de nav/footer se répercute
//  dans les 3 fichiers"). Appelé en hook PostToolUse après Edit/Write.
//
// Implémentation de RÉFÉRENCE (Node). Un miroir Python existe
// (check-nav-consistency.py) — si tu modifies l'une, modifie l'autre.
//
// Entrée : JSON du hook sur stdin (contient tool_input.file_path).
// Sortie : exit 0 = cohérent / non concerné ; exit 2 = divergence
//          (le message stderr est remonté à Claude pour correction).
// ===========================================================
import { readFileSync } from 'node:fs';
import { basename, dirname, join } from 'node:path';

const PAGES = ['index.html', 'projets.html', 'experiences.html'];

// --- 1. Lire l'entrée du hook (tolérant : exécution manuelle possible) ---
let hook = {};
try { hook = JSON.parse(readFileSync(0, 'utf8') || '{}'); } catch { /* pas de stdin */ }
const editedPath = hook?.tool_input?.file_path || '';

// Ne se déclenche que si le fichier édité est l'une des 3 pages.
if (editedPath && !PAGES.includes(basename(editedPath))) process.exit(0);

// --- 2. Racine du repo : env du hook, sinon dossier du fichier édité, sinon cwd ---
const root = process.env.CLAUDE_PROJECT_DIR
  || (editedPath ? dirname(editedPath) : process.cwd());

// --- 3. Extraction ---
const stripTags = (s) => s.replace(/<[^>]*>/g, ' ').replace(/\s+/g, ' ').trim();

function extract(html) {
  const nav = (html.match(/<nav\b[^>]*>([\s\S]*?)<\/nav>/i) || [, ''])[1];
  const logo = stripTags((nav.match(/<a\b[^>]*class="[^"]*nav-logo[^"]*"[^>]*>([\s\S]*?)<\/a>/i) || [, ''])[1]);
  const linksBlock = (nav.match(/<ul\b[^>]*class="[^"]*nav-links[^"]*"[^>]*>([\s\S]*?)<\/ul>/i) || [, ''])[1];
  const labels = [...linksBlock.matchAll(/<a\b[^>]*>([\s\S]*?)<\/a>/gi)].map(m => stripTags(m[1]));
  const footer = stripTags((html.match(/<footer\b[^>]*>([\s\S]*?)<\/footer>/i) || [, ''])[1]);
  return { logo, labels, footer };
}

// --- 4. Lire les 3 pages ---
const data = {};
for (const p of PAGES) {
  try { data[p] = extract(readFileSync(join(root, p), 'utf8')); }
  catch { process.exit(0); } // une page manque -> pas de comparaison possible
}

// --- 5. Comparer par rapport à index.html (référence) ---
const ref = data['index.html'];
const problems = [];
for (const p of PAGES.slice(1)) {
  const d = data[p];
  if (d.logo !== ref.logo)
    problems.push(`- Logo différent dans ${p} : "${d.logo}" vs "${ref.logo}" (index.html)`);
  if (JSON.stringify(d.labels) !== JSON.stringify(ref.labels))
    problems.push(`- Libellés de nav différents dans ${p} :\n    ${p} = [${d.labels.join(', ')}]\n    index.html = [${ref.labels.join(', ')}]`);
  if (d.footer !== ref.footer)
    problems.push(`- Footer différent dans ${p} :\n    ${p} = "${d.footer}"\n    index.html = "${ref.footer}"`);
}

if (problems.length) {
  console.error('[check-nav-consistency] Divergence nav/footer entre les pages (règle CLAUDE.md #2) :\n'
    + problems.join('\n')
    + '\nRépercute le changement de nav/footer dans les 3 fichiers (index.html, projets.html, experiences.html).');
  process.exit(2);
}
process.exit(0);

File: .claude/hooks/check-nav-consistency.py
# ===========================================================
# Garde-fou : cohérence nav / footer entre les 3 pages HTML.
# Miroir de check-nav-consistency.mjs (Node = référence).
# Si tu modifies l'un, modifie l'autre.
#
# Entrée : JSON du hook sur stdin (tool_input.file_path).
# Sortie : exit 0 = cohérent / non concerné ; exit 2 = divergence.
# ===========================================================
import sys, os, json, re

PAGES = ['index.html', 'projets.html', 'experiences.html']

# --- 1. Entrée du hook (tolérant à une exécution manuelle) ---
try:
    hook = json.loads(sys.stdin.read() or '{}')
except Exception:
    hook = {}
edited = (hook.get('tool_input') or {}).get('file_path', '')

if edited and os.path.basename(edited) not in PAGES:
    sys.exit(0)

# --- 2. Racine du repo ---
root = os.environ.get('CLAUDE_PROJECT_DIR') \
    or (os.path.dirname(edited) if edited else os.getcwd())

# --- 3. Extraction ---
def strip_tags(s):
    return re.sub(r'\s+', ' ', re.sub(r'<[^>]*>', ' ', s)).strip()

def extract(html):
    nav = (re.search(r'<nav\b[^>]*>([\s\S]*?)</nav>', html, re.I) or [None, ''])
    nav = nav.group(1) if hasattr(nav, 'group') else ''
    logo_m = re.search(r'<a\b[^>]*class="[^"]*nav-logo[^"]*"[^>]*>([\s\S]*?)</a>', nav, re.I)
    logo = strip_tags(logo_m.group(1)) if logo_m else ''
    links_m = re.search(r'<ul\b[^>]*class="[^"]*nav-links[^"]*"[^>]*>([\s\S]*?)</ul>', nav, re.I)
    links = links_m.group(1) if links_m else ''
    labels = [strip_tags(m) for m in re.findall(r'<a\b[^>]*>([\s\S]*?)</a>', links, re.I)]
    foot_m = re.search(r'<footer\b[^>]*>([\s\S]*?)</footer>', html, re.I)
    footer = strip_tags(foot_m.group(1)) if foot_m else ''
    return {'logo': logo, 'labels': labels, 'footer': footer}

# --- 4. Lire les 3 pages ---
data = {}
for p in PAGES:
    try:
        with open(os.path.join(root, p), encoding='utf-8') as f:
            data[p] = extract(f.read())
    except Exception:
        sys.exit(0)

# --- 5. Comparer par rapport à index.html ---
ref = data['index.html']
problems = []
for p in PAGES[1:]:
    d = data[p]
    if d['logo'] != ref['logo']:
        problems.append(f'- Logo différent dans {p} : "{d["logo"]}" vs "{ref["logo"]}" (index.html)')
    if d['labels'] != ref['labels']:
        problems.append(f'- Libellés de nav différents dans {p} :\n    {p} = {d["labels"]}\n    index.html = {ref["labels"]}')
    if d['footer'] != ref['footer']:
        problems.append(f'- Footer différent dans {p} :\n    {p} = "{d["footer"]}"\n    index.html = "{ref["footer"]}"')

if problems:
    sys.stderr.write('[check-nav-consistency] Divergence nav/footer entre les pages (règle CLAUDE.md #2) :\n'
                     + '\n'.join(problems)
                     + '\nRépercute le changement de nav/footer dans les 3 fichiers.\n')
    sys.exit(2)
sys.exit(0)
